Sources/Plasma/CoreLib/hsBitVector.h

```diff
@@ -116,137 +116,114 @@ class hsBitVector {
             fNumBitVectors = other.fNumBitVectors;
             fBitVectors = new uint32_t[fNumBitVectors];
-        }
-        else
-        {
+        } else {
             Clear();
         }
 
-        int i;
-        for( i = 0; i < other.fNumBitVectors; i++ )
+        for (uint32_t i = 0; i < other.fNumBitVectors; i++)
             fBitVectors[i] = other.fBitVectors[i];
     }
     return *this;
 }
 
 inline bool hsBitVector::operator==(const hsBitVector& other) const
 {
-    if( fNumBitVectors < other.fNumBitVectors )
+    if (fNumBitVectors < other.fNumBitVectors)
         return other.operator==(*this);
-    int i;
-    for( i = 0; i < other.fNumBitVectors; i++ )
-        if( fBitVectors[i] ^ other.fBitVectors[i] )
+    uint32_t i;
+    for (i = 0; i < other.fNumBitVectors; i++)
+        if (fBitVectors[i] ^ other.fBitVectors[i])
             return false;
-    for( ; i < fNumBitVectors; i++ )
-        if( fBitVectors[i] )
+    for (; i < fNumBitVectors; i++)
+        if (fBitVectors[i])
             return false;
     return true;
 }
 
 inline hsBitVector& hsBitVector::operator&=(const hsBitVector& other)
 {
-    if( this == &other )
+    if (this == &other)
         return *this;
 
-    if( fNumBitVectors > other.fNumBitVectors )
-    {
+    if (fNumBitVectors > other.fNumBitVectors)
         fNumBitVectors = other.fNumBitVectors;
-    }
-    int i;
-    for( i = 0; i < fNumBitVectors; i++ )
+    for (uint32_t i = 0; i < fNumBitVectors; i++)
         fBitVectors[i] &= other.fBitVectors[i];
     return *this;
 }
 
 inline hsBitVector& hsBitVector::operator|=(const hsBitVector& other)
 {
-    if( this == &other )
+    if (this == &other)
         return *this;
 
-    if( fNumBitVectors < other.fNumBitVectors )
-    {
+    if (fNumBitVectors < other.fNumBitVectors)
         IGrow(other.fNumBitVectors);
-    }
-    int i;
-    for( i = 0; i < other.fNumBitVectors; i++ )
+    for (uint32_t i = 0; i < other.fNumBitVectors; i++)
         fBitVectors[i] |= other.fBitVectors[i];
     return *this;
 }
 
 inline hsBitVector& hsBitVector::operator^=(const hsBitVector& other)
 {
-    if( this == &other )
-    {
+    if (this == &other) {
         Clear();
         return *this;
     }
 
-    if( fNumBitVectors < other.fNumBitVectors )
-    {
+    if (fNumBitVectors < other.fNumBitVectors)
         IGrow(other.fNumBitVectors);
-    }
-    int i;
-    for( i = 0; i < other.fNumBitVectors; i++ )
+    for (uint32_t i = 0; i < other.fNumBitVectors; i++)
         fBitVectors[i] ^= other.fBitVectors[i];
     return *this;
 }
 
 inline hsBitVector& hsBitVector::operator-=(const hsBitVector& other)
 {
-    if( this == &other )
-    {
+    if (this == &other) {
         Clear();
         return *this;
     }
 
-    int minNum = fNumBitVectors < other.fNumBitVectors ? fNumBitVectors : other.fNumBitVectors;
-    int i;
-    for( i = 0; i < minNum; i++ )
+    uint32_t minNum = fNumBitVectors < other.fNumBitVectors ? fNumBitVectors : other.fNumBitVectors;
```

Comment:
`std::min`

Reply:
Do we want to include the algorithm header in this header?

Reply:
Nah, probably not worth it here...